### ISSUE TITLE
Once Blocked can the application be removed ? 

### ISSUE THREAD

**Issue by GenoMyb** [hjuutilainen/bigsurblocker#4]
Once the application has been blocked can It be deleted ?
eg I have added the bundleID fro AnyDesk
Once the warning has been displayed to the users and the Users clicks ok the Application gets deleted

**Comment by porteusconf** [comment]
I'm just an end user, but if you are asking if you can remove bigsurblocker completely, so as to be able to run "Install Big Sur.app" (or whatever it is called), that is easy: you  can disable it, then just delete the only 2 files it installs,  with these commands in Terminal.
``` shell
$  launchctl unload /Library/LaunchAgents/com.hjuutilainen.bigsurblocker.plist  # disables bigsurblocker
$  sudo  rm  /Library/LaunchAgents/com.hjuutilainen.bigsurblocker.plist
$  sudo  rm  /usr/local/bin/bigsurblocker
```
But if you are asking if you can block the downloading of the multi-gig big-sur installer app in the first place, then maybe the easiest way is to Un-check "Download new updates when available" in SysPrefs:Softwareupdate:Advanced. There may be ways to block the download of only the bigsur installer(s), and I'll update this comment if I figure that out.

**Comment by porteusconf** [comment]
As of today, Thur Nov 12, when Big Sur is going to start becoming available, these 2 commands would protect you:
```
/usr/bin/curl  -LJO  https://github.com/hjuutilainen/bigsurblocker/releases/download/v20201111/bigsurblocker-20201111.pkg

sudo installer -pkg  ./bigsurblocker-20201111.pkg   -target / 
```
Above is for version 20201111 so If/when  there is a newer release, use newer download link. It looks like this for me:
```
$ cd /private/tmp  ## everything in /private/tmp is automatically deleted during every reboot
$ /usr/bin/curl  -LJO  https://github.com/hjuutilainen/bigsurblocker/releases/download/v20201111/bigsurblocker-20201111.pkg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   652  100   652    0     0   8596      0 --:--:-- --:--:-- --:--:--  8693
100 48046  100 48046    0     0   205k      0 --:--:-- --:--:-- --:--:--  205k
curl: Saved to filename 'bigsurblocker-20201111.pkg'
$ sudo installer -pkg  ./bigsurblocker-20201111.pkg   -target /
Password:
installer: Package name is bigsurblocker-20201111
installer: Upgrading at base path /
installer: The upgrade was successful.
$ 
```
No need to clean up if you do a `cd /private/tmp` first, as shown; everything in /tmp is deleted automatically at every reboot.

**Comment by Vaultnaemsae** [comment]
OK, total newb here. How do you uninstall this? I copied and pasted the uninstall command in terminal as directed:

```
current_user_uid=$( echo "show State:/Users/ConsoleUser" | scutil | awk '/UID :/ && ! /loginwindow/ { print $3 }' )

launchd_item_path="/Library/LaunchAgents/com.hjuutilainen.bigsurblocker.plist"
launchctl bootout gui/${current_user_uid} "${launchd_item_path}"

rm -f /Library/LaunchAgents/com.hjuutilainen.bigsurblocker.plist
rm -f /usr/local/bin/bigsurblocker

pkgutil --forget com.hjuutilainen.bigsurblocker
```
 Doesn't work. say I don't have permission. Can anybody help? I don't know what "(as root or with sudo)" means.

**Comment by hjuutilainen** [comment]
@Vaultnaemsae If this is your own computer and you're the only user, you can simplify the uninstall. If you have an admin account you can copy and paste the following commands to Terminal.app and reboot your computer. The second command starting with `sudo` will ask for your password.

```
launchctl unload /Library/LaunchAgents/com.hjuutilainen.bigsurblocker.plist
sudo rm -f /Library/LaunchAgents/com.hjuutilainen.bigsurblocker.plist
sudo rm -f /usr/local/bin/bigsurblocker
sudo pkgutil --forget com.hjuutilainen.bigsurblocker
```

**Comment by securingmom** [comment]
>  There may be ways to block the download

@porteusconf 

Can a file be touched and creative permissions be applied to prevent the download of the same name overwriting it?

**Comment by najoey** [comment]
Do you have a uninstall .pkg created from the script. signed or unsigned?

**Comment by sensaetions** [comment]
Hello @hjuutilainen,

I followed all of the instructions to uninstall/remove the BigSurBlocker. However, I'm still not able to see Big Sur as the OS option to install anymore. Every time I use Mac Software Update to try to find Big Sur on App Store, it says "The requested version of macOS is not available."

What else can I do? How do I know that the BigSurBlocker has been successfully removed?

**Comment by hjuutilainen** [comment]
Big Sur Blocker should not (and can not) block the visibility of Big Sur in System Preferences -> Software Update due to the restrictions that Apple is using. The update should appear there even if Big Sur Blocker is running and it’s only after downloading that we can prevent the Install macOS Big Sur.app from starting. 

Try ensuring that the /Library/LaunchAgents/com.hjuutilainen.bigsurblocker.plist file is deleted and reboot your computer. Then after the reboot, immediately start System Preferences -> Software Update.

**Comment by sensaetions** [comment]
@hjuutilainen 

Yes, I followed all of the uninstall instructions. There is no more file by the name of `com.hjuutilainen.bigsurblocker.plist` on my computer. I've restarted multiple times, and Software Update cannot find the OS updates.

Something isn't working correctly. Now my Mac only recognizes Catalina, and even that refuses to fully download -- it's indicating "null" -- and stops the download process. 

Any ideas on a fix?

**Comment by posguy99** [comment]
Bigsurblocker kills the installer if it starts.  If doesn't do anything to stop Software Update from seeing the installer.  

-- 
 Marc Wilson

**Comment by sensaetions** [comment]
@hjuutilainen @posguy99

Any ideas then what could be causing the "null" messaging and Software Update's inability to see any further updates, even after uninstalling BigSurBlocker?

**Comment by sensaetions** [comment]
@hjuutilainen @posguy99

Any ideas then what could be causing the "null" messaging and Software Update's inability to see any further updates, even after uninstalling BigSurBlocker? Big Sur was showing before installing BigSurBlocker. Now my Mac won't even install Catalina update.



> Bigsurblocker kills the installer if it starts. If doesn't do anything to stop Software Update from seeing the installer.
> […](#)
> -- Marc Wilson
> On Jun 4, 2021, at 11:19 AM, sensaetions ***@***.***> wrote: ﻿ @hjuutilainen Yes, I followed all of the uninstall instructions. There is no more file by the name of com.hjuutilainen.bigsurblocker.plist on my computer. I've restarted multiple times, and Software Update cannot find the OS updates. Something isn't working correctly. Now my Mac only recognizes Catalina, and even that refuses to fully download -- it's indicating "null" -- and stops the download process. Any ideas on a fix? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by sensaetions** [comment]
Updating: 
I also did a terminal command `sudo softwareupdate --reset-ignored` to ignore any software update blockers. This allowed me to update my macOS as expected.

**Comment by mrsallad** [comment]
@sensaetions exactly the same for me. Terminal command solved it for now. If you have any updates in the future about why this happened please let me know.

**Comment by pfuth** [comment]
> sudo softwareupdate --reset-ignored

Is this the expected result

`Ignored updates:`
`(`
`)`
``
`Ignoring software updates is deprecated.`
`The ability to ignore individual updates will be removed in a future release of macOS. `
